### PR TITLE
Fix various races

### PR DESCRIPTION
**Pull request by usmanong**
1. Race during target runner start and stats runner log
2. Race during proxy start and keepalive runner
3. Race during target/proxy start and keepalive's first register call

@VladimirMarkelov @liangdrew @sasanap @alex-aizman

### CODE REVIEW

File: dfc/keeper.go [hunk 1]
```diff
@@ -208,11 +208,15 @@ func (r *proxykalive) keepalive(err error) (stopped bool) {
...
+
 	if r.p.primary {
+		smapLock.Unlock()
 		return r.primarykeepalive()
 	}
 
 	if r.p.smap.ProxySI == nil || !r.timedOut(r.p.smap.ProxySI.DaemonID) {
+		smapLock.Unlock()
 		return
 	}
```
Comment (VladimirMarkelov) [inline dfc/keeper.go]: Why smapLock is not Unlocked here?

**Comment by usmanong** [comment]
actually hod on to this review, let me run all tests, sorry forgot to run it

File: dfc/keeper.go (hunk 1 above)
Comment (usmanong) [reply dfc/keeper.go]: yup, that's why i think i need to run the whole tests, i only verified that all daemon starts
great eyes, that's why my code need reviews, thanks

**Comment by usmanong** [comment]
ok, please review, cleared all tests
thanks

**Pull request merged**